src/http.js

```diff
@@ -110,87 +116,81 @@ Zotero.HTTP = new function() {
 		}
 		Zotero.debug(`HTTP ${method} ${requestURL}${logBody}`);
 		
-		return new Promise(function(resolve, reject) {
-			request({
-				uri: requestURL,
-				method,
-				headers: options.headers,
-				timeout: options.timeout,
-				body: options.body,
-				gzip: true,
-				followAllRedirects: true,
-				jar: options.cookieSandbox
-			}, function(error, response, body) {
-				if (error) {
-					return reject(error);
-				}
-				
-				// Array of success codes given
-				if (options.successCodes) {
-					var success = options.successCodes.includes(response.statusCode);
-				}
-				// Explicit FALSE means allow any status code
-				else if (options.successCodes === false) {
-					var success = true;
-				}
-				// Otherwise, 2xx is success
-				else {
-					var success = response.statusCode >= 200 && response.statusCode < 300;
-				}
-				if (!success) {
-					return reject(new Zotero.HTTP.StatusError(requestURL, response.statusCode, response.body));
-				}
+		let {response, body} = await customRequest(method, requestURL, options);
+		
+		if (!response.headers['content-type']) {
+			return reject(new Error('Missing content-type header'));
```

Comment:
For this we should also return 400, since we don't want to repeat translation too. It could be something like `InvalidResponse`. But maybe it's better to wait until we merge #59 because I suspect some changes for `content-type` filtering.